server/src/main/java/org/opensearch/wlm/tracker/QueryGroupResourceUsageTrackerService.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.tracker;
+
+import org.opensearch.search.ResourceType;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskResourceTrackingService;
+import org.opensearch.wlm.QueryGroupLevelResourceUsageView;
+import org.opensearch.wlm.QueryGroupTask;
+
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * This class tracks resource usage per QueryGroup
+ */
+public class QueryGroupResourceUsageTrackerService {
```

Comment:
It looks like this service isn't initialized anywhere, I'm guessing that's coming in a future PR?

Reply:
That's correct. 
This is just tracking related changes, I have a followup PR for cancellation. 
I will then create a PR connecting both tracking and cancellation.